### ISSUE TITLE
Monolog - loglevel changes to 'debug' and ignores legacy settings

### ISSUE THREAD

**Issue by AndyHee** [friendica/friendica#6382]
Fail2ban cannot read the friendica log any more. My fail2ban log now reports this:

`2019-01-04 23:13:01,595 fail2ban.filter         [2307]: WARNING Error decoding line from '/var/www/friendica/logs/friendica.log' with 'utf-8'. Consider setting logencoding=utf-8 (or another appropriate encoding) for this jail. Continuing to process line ignoring invalid characters: 
`

The fail2ban log has become massive and unmanageable for human readers.

### Friendica version you encountered the problem
develop

**Comment by AndyHee** [comment]
Configurations for `/etc/fail2ban/jail.local` include these lines

```
[friendica]
logencoding = utf-8
```

**Comment by AndyHee** [comment]
No problem. I'll send you a PM as soon as I have access to my server.

**Comment by nupplaphil** [comment]
Thanks, I got the PM!
I created both log-types on my server and their encoding are identical at first sight (UTF-8). The line endings too (LF).

Did you set the `logencoding = utf-8` after the error message appears or before?

**Comment by tobiasd** [comment]
I think this is a bug, as before fail2ban could read the friendica log.

**Comment by AndyHee** [comment]
> Did you set the logencoding = utf-8 after the error message appears or before?

Before. I think, it's the standard setting for Friendica as specified in the install instructions.

**Comment by AndyHee** [comment]
@nupplaphil did the new monolog feature change the logging level?

After pulling this new feature my friendica log grew to several hundred M in size within a few hours. Could it be that the logging level was changed to "raw" ? My log level is normally set to "warn" and that's why content turned up in my friendica log that fail2ban had never decoded before.

**Comment by nupplaphil** [comment]
Ha, you're right! Please set the `loglevel` manually to `notice` (now string, not int anymore).

**Comment by AndyHee** [comment]
Thanks @nupplaphil ! How do I set the loglevel manually? Can I already use the gui in the admin panel?

**Comment by AndyHee** [comment]
Thanks a lot, @nupplaphil ! 

I had reset my installation. I will git pull again now and try to see what happens to the original logs settings.

**Comment by AndyHee** [comment]
OK, I think I know what happened.

Pulling this new feature without setting the log level manually does not work. My level changed from "Warn" to "Debug" (or higher?) as soon as I pulled. This consequently caused an unusually/ unexpectedly high strain on the system.

The fail2safe warning is just incidental and likely the result of the fire-hosed friendica log.

If this makes sense and you agree with my reasoning, I'll change the title of the issue.

**Comment by AndyHee** [comment]
Now this is what I have in `local.ini.php`

```
[system]
loglevel = 'error'

```

But I level is still on debugging and my fail2ban log is filling up with warning messages and more importantly fail2ban seems not working for the friendica jail.

I have restarted myslq, nginx, php7.3-fm but it seems to have no effect on the log level. What I'm I missing?

**Comment by nupplaphil** [comment]
.. this is strange .. I added the same loglevel and and I don't get any errors ..

my test `local.ini.php`:
```php
[system]
debugging = true
logfile = friendica.log
loglevel = 'error'
```

I added a custom `Logger::error('test');` to `index.php` and it appears, but no other notice/info. When I change to `loglevel = 'debug'` I get a lot of other notices too .. @MrPetovan do you have any idea what I additionally can check?

@AndyHee but it seems to be an loglevel issue, so I'd change the title according to it.

**Comment by nupplaphil** [comment]
@AndyHee An idea: Is there an entry in the `config` table for `system.loglevel` ? Maybe this is overriding the current setting. Because I added it for #6396  ..

**Comment by AndyHee** [comment]
@nupplaphil there is no such entry in the DB; I'm still on version `1295`.

Whatever I add to `local.ini.php` in terms of the three values as shown https://github.com/friendica/friendica/issues/6382#issuecomment-451742257 has no effect so ever. It just keeps logging at debug level.

I revert back for now and see what happens with your next PR. It is possibly related to my setup. @copiis mentioned that he turned off logging and he has already pulled this feature in. Perhaps he can confirm that it works at last in terms of disabling logging.

**Comment by MrPetovan** [comment]
Youcan check the actual value of any config key by running `$> bin/console config system loglevel`. If it's different than what you set in the config file, you can also set the value with `$> bin/console config system loglevel error`.

**Comment by AndyHee** [comment]
Thanks @MrPetovan ! Initially the level was "0"  I set it to `error` now via this command and it still has no effect.

**Comment by MrPetovan** [comment]
> Once logging is in full progress, any changes to loglevel will not be immediate. Changing the loglevel via the console can take several minutes to take effect. However, it (eventually) works as @MrPetovan suggested.

This is because the background task runs in memory and spawns many more threads sharing the same memory before dying and being replaced y a fresh copy with the updated config values. It was the case with the previous logger levels as well, there would be several minutes during which the background task would keep using the old level.

**Comment by AndyHee** [comment]
Thanks for the explanation! :-) [Signing off now.]

**Comment by AndyHee** [comment]
I've updated the title (again), as the fail2ban issues is _not_ related to the automatic changes in the loglevel. Please see here: https://github.com/friendica/friendica/issues/6403 .

**Comment by nupplaphil** [comment]
Got it @AndyHee !

I checked if the value is an integer before mapping legacy 0 - 5. But `Config::get()` returns **every** value as a string, so it skipped the mapping. And later monolog doesn't know what "0" means and enables `debug` as fallback.

Should be fixed with #6409

**Comment by AndyHee** [comment]
Great! I close now.
